core/src/player.rs

```diff
@@ -1441,6 +1494,27 @@ impl Player {
         needs_render
     }
 
+    ///This searches for a display object by it's id
+    //TODO: is there a better place to place next two functions
+    fn find_first_character_instance<'gc>(
+        obj: DisplayObject<'gc>,
+        character_id: &CharacterId,
+    ) -> Option<DisplayObject<'gc>> {
+        if let Some(parent) = obj.as_container() {
+            for child in parent.iter_render_list() {
+                if &child.id() == character_id && Arc::ptr_eq(&child.movie(), &obj.movie()) {
```

Comment:
While this is clever, it won't work, because we specifically need to check `child`'s movie against the movie that defined `character_id`, not the movie that defined it's parent. As it stands this code will break on the following display tree configuration:

 * Level 0 / root movie clip (Movie A, Character 0)
     * Button A (Movie A, Character 1)
     * Child movie's root clip (Movie B, Character 0)
       * Button B (Movie B, Character 1)

Calling `find_first_character_instance` looking for Button B will instead return Button A, because:

 * Button A and Level 0 are both defined in Movie A
 * Button A is character ID 1

You specifically need to provide the movie that defined the character ID you're interested in here as a separate function parameter.